RandLAPACK/drivers/rl_revd2.hh

```diff
@@ -0,0 +1,240 @@
+#ifndef randlapack_drivers_revd2_h
+#define randlapack_drivers_revd2_h
+
+#include "rl_qb.hh"
+#include "rl_blaspp.hh"
+#include "rl_lapackpp.hh"
+#include "rl_util.hh"
+
+#include <RandBLAS.hh>
+#include <cstdint>
+#include <vector>
+
+namespace RandLAPACK {
+
+template <typename T>
+class REVD2alg {
+    public:
+
+        virtual ~REVD2alg() {}
+
+        virtual int call(
+            int64_t m,
+            std::vector<T>& A,
+            int64_t& k,
+            std::vector<T>& V,
+            std::vector<T>& E
+        ) = 0;
+};
+
+template <typename T>
+class REVD2 : public REVD2alg<T> {
+    public:
+
+        // Constructor
+        REVD2(
+            RandLAPACK::RangeFinder<T>& rf_obj,
+            T tolerance,
+            int power_iters,
+            RandBLAS::base::RNGState<r123::Philox4x32> s,
+            bool verb
+        ) : RF_Obj(rf_obj) {
+            tol = tolerance;
+            p = power_iters;
+            state = s;
+            verbosity = verb;
+        }
+
+        /// Computes a rank-k approximation to an EVD of a symmetric positive definite matrix:
+        ///     A= V diag(E) V*,
+        /// where V is a matrix of eigenvectors and E is a vector of eigenvalues.
+        /// Does so adaptively. 
+        /// If the tolerance is not met, increases the rank estimation parameter by 2.
+        /// Tolerance is the maximum of user-specified tol times 5 and computed 'nu' times 5.
+        /// This is motivated by the fact that the approx error will never be smaller than nu.
+        /// This code is identical to ALgorithm E2 from https://arxiv.org/pdf/2110.02820.pdf.
+        /// Uses RangeFinder for constructing a sketching operator.
+        /// Has a lot of potential in terms of storage space optimization, 
+        /// which, however, will affect readability.
+        ///
+        /// @param[in] m
+        ///     The number of rows in the matrix A.
+        ///
+        /// @param[in] n
+        ///     The number of columns in the matrix A.
+        ///
+        /// @param[in] A
+        ///     The m-by-m matrix A, stored in a column-major format.
+        ///     Must be SPD.
```

Comment:
Should be specified by its upper or lower triangle.

Reply:
I was actually assuming that A is stored fully.

Reply:
I think it's best to not make that assumption. Users with Hermitian / symmetric matrices already expect to work with only one of the triangular parts. It's true that the memory in the opposite triangle is just sitting there, but the user might have reserved that for some kind of workspace.

Having the user specify a triangle also reduces the likelihood of the code being called on malformed input. For example, the call to Cholesky [here](https://github.com/BallisticLA/RandLAPACK/pull/35/files#diff-01caf3ea6e344fc58f135bc5fdaed76173d08a13509a5ef7290bd6741eb4c014R156) might succeed even if ``R`` isn't SPD, so long as the symmetrization of R is SPD. We wouldn't want that to happen.

Reply:
Ran into an issue while doing this & idk why it is happening. May need help.

Reply:
Fixed.